### PR TITLE
Vol geom wasm texture update

### PR DESCRIPTION
**Pull request by capnramses**
## Purpose of this PR

* PR allows WASM to get embedded textures from sequence files using vol_geom (vol_av not needed in this case).

## Changes made in this PR

* Wrote a separate demo to test changes to WASM code, using Basis Universal (.basis) embedded textures in a new OpenGL program and a new WebGL demo (not included in this PR).
* These changes are just an API update, and are non-breaking changes. Example using new API to come in subsequent PR.
* Updates the pre.js script -- please check this still works with existing examples in case I broke something!!!
* Also updated example 5 per request of @patrick-volograms -- corrected path to video file.

## Testing Summary

* Used the sample 'California' vologram which I re-encoded for embedded textures.
* Also used a new re-encoded 'Rafa' vologram from Jan, further validating the code.

image

- [x] Explained how reviewers build, deploy, and test your code and what should they look for that would indicate a bug or regression.
- [x] Tested locally. Which platforms? OpenGL on Windows and Linux, WebGL on Windows, Linux, MacOS desktops, iPhone and Android.
- [x] Stated supported platforms have you not tested on, that you need help with? -- none for now!
- [x] Named specific tools have you ran over the code: e.g. which { compilers, debuggers, fuzzers, profilers, sanitisers, linters, other tools }. emcc, gcc, clang, I also used a javascript error checker.
- [ ] Ran any unit tests that cover this code. No - this is a weak point - need to check changes manually in pre, and later will validate wasm with a working demo.
- [x] Explained inputs used to test for bugs. e.g. existing known working data, large files, empty files, etc.

## Risk Introduced and any Breaking Changes

Provide more detail here as required.

- [ ] This PR introduces breaking changes (API or ABI). Instructions are provided for existing users to update their projects.
- [ ] This PR introduces new dependencies. You have provided links and licences in this section. This is **_extremely important!_**.
- [ ] New dependencies: Explained how any new dependencies are linked and built againt - .dll, inline code etc.?
- [x] Gaps in testing: Is any code not covered by tests? Changes to pre.js.
- [ ] Gaps in testing: Are you unable to test on some platform? Or is some input data not available to you for testing?
- [ ] Analysed any other potential risks here.

## Pre-Merge Checklist

* If you have a problem with the review process (e.g. it's taking too long, or the reviewers forgot) get in touch with the Team Lead of the project!

- [x] Before making your PR, refresh your branch with a merge from its parent in case of conflicting changes since you started your branch.
- [x] This PR description is written clearly so that reviewers can understand the why/what/how of the work done.
- [x] Reviewed your own PR and checked your changed code.
- [x] Invited a range of good reviewers.
- [x] Wait for a PR to be properly reviewed. Responded to all questions, and made any required changes.
- [x] Received one other reviewer's approval. Ideally wait for 2+ approvals, especially if you didn't get a thorough reivew on the first approval.
- [x] If a reviewer asks a question, even if you have another approval, got the approval of the other reviewers as well before merging.
- [x] Made sure existing reviewers approved any changes to the PR.
- [x] Resolved any merge conflicts after making the PR.
- [x] CI/CD builds are not reporting any errors.

### CODE REVIEW

File: wasm/pre.js [hunk 1]
```diff
@@ -41,6 +44,12 @@ Module.initVologramFunctions = () => {
     Module['frame_normals_sz'] = Module.cwrap(
         'frame_normals_sz', 'number'
     );
+    Module['frame_texture_data_ptr'] = Module.cwrap(
```
Comment (capnramses) [inline wasm/pre.js:47]: @patrick-volograms In my code I'm using these data pointers as 'number' (I feel like I read something like that in the emscripten docs) but I notice the other data ptr here has 'array' type. I'm not sure if that matters or what the difference is?

File: wasm/wasm_vol_geom.c [hunk 2]
```diff
@@ -43,7 +54,7 @@ EMSCRIPTEN_KEEPALIVE
 bool free_file_info( void ) { return vol_geom_free_file_info( &_info ); }
 
 EMSCRIPTEN_KEEPALIVE
-int32_t frame_count( void ) { return _info.hdr.frame_count; }
+int32_t frame_count( void ) { return (int32_t)_info.hdr.frame_count; }
```
Comment (capnramses) [inline wasm/wasm_vol_geom.c:57]: pedantic cast to keep compiler happy

File: examples/05_vol_geom_wasm_threejs/index.html [hunk 3]
```diff
@@ -52,7 +52,7 @@
 	var vologram = new Object();   // Stores state about the current vologram.
 	vologram.header_url = "./calif/header.vols";
 	vologram.sequence_url = "./calif/sequence_0.vols";
-	vologram.video_url = "./calif/output.mp4";
+	vologram.video_url = "./calif/output_720.mp4";
```
Comment (capnramses) [inline examples/05_vol_geom_wasm_threejs/index.html:55]: i think this is correct @patrick-volograms ?

Comment (patrick-volograms) [reply examples/05_vol_geom_wasm_threejs/index.html:55]: I changed the structure of the examples in my last PR (sorry, forgot to tell you). I have all the examples referencing the same vologram in a different folder now.

**Review by patrick-volograms — APPROVED** [review]

**Pull request merged**